lib/rules/indent.js

```diff
@@ -521,540 +393,486 @@ module.exports = {
-                            parent.type === "CallExpression" ||
-                            parent.type === "ArrowFunctionExpression" ||
-                            parent.type === "NewExpression" ||
-                            parent.type === "LogicalExpression"
-                        ) {
-                            nodeIndent = nodeIndent + indentSize;
-                        }
-                    }
-                } else if (!parentVarNode && !isFirstArrayElementOnSameLine(parent) && parent.type !== "MemberExpression" && parent.type !== "ExpressionStatement" && parent.type !== "AssignmentExpression" && parent.type !== "Property") {
-                    nodeIndent = nodeIndent + indentSize;
+        /**
+        * Check indentation for lists of elements (arrays, objects, function params)
+        * @param {Token[]} tokens list of tokens
+        * @param {ASTNode[]} elements List of elements that should be offset
+        * @param {number} offset The amount that the elements should be offset
+        * @returns {void}
+        */
+        function addElementListIndent(tokens, elements, offset) {
+
+            /**
+            * Gets the first token of an element at the given index, including surrounding parentheses.
+            * @param {number} elementIndex The index of the element in `elements`
+            * @returns {Token} The first token of this element
+            */
+            function getFirstTokenByElementsIndex(elementIndex) {
+                let token = sourceCode.getTokenBefore(sourceCode.getFirstToken(elements[elementIndex]));
+
+                while (token.type === "Punctuator" && token.value === "(" && token !== tokens[0]) {
+                    token = sourceCode.getTokenBefore(token);
                 }
 
-                elementsIndent = nodeIndent + indentSize;
-
-                checkFirstNodeLineIndent(node, nodeIndent);
-            } else {
-                nodeIndent = getNodeIndent(node).goodChar;
-                elementsIndent = nodeIndent + indentSize;
+                return sourceCode.getTokenAfter(token);
             }
 
-            /*
-             * Check if the node is a multiple variable declaration; if so, then
-             * make sure indentation takes that into account.
-             */
-            if (isNodeInVarOnTop(node, parentVarNode)) {
-                elementsIndent += indentSize * options.VariableDeclarator[parentVarNode.parent.kind];
+            if (offset === "first" && elements.length) {
+                ignoreToken(getFirstTokenByElementsIndex(0));
             }
 
-            checkNodesIndent(elements, elementsIndent);
+            // Run through all the tokens in the list, and offset them by one indent level (mainly for comments, other things will end up overridden)
+            setDesiredOffsets(tokens.slice(1, -1), tokens[0], offset === "first" ? 1 : offset);
+            elements.slice(1).forEach((element, index) => {
```

Comment:
Not a huge deal, but the `index + 1` constructs in this callback are entirely due to the initial slice, right? I don't know if it'd be more readable or less to remove the slice and add `if (index === 0) { return; }` to the beginning of the callback or not... Or even just adding a `const actualIndex = index + 1;` or something.

Just something to think about.